### PR TITLE
Fix JS unit tests

### PR DESCRIPTION
**Pull request by chihsuan**
### Changes proposed in this Pull Request:

Fixes the broken JS unit tests.

- Add jest and babel configs to add necessary mocks and setup files in the bridge.
- Add `@testing-library/jest-dom` and `@testing-library/react` libs as we use them in our tests.
- Add test mock and setup files
- Fix wrong `enqueueScript` import path in payment gateway test

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Run `nvm use` - if you don't have version 16 of node installed, you may need to run nvm install 16
2. Run `npm install`
3. Run `npm run test:unit`
4. Confirm all tests pass
5. Run `npm run build`
6. Verify that the build completes successfully

Screenshot 2023-04-25 at 11 57 12

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by rjchow — COMMENTED** [review]
I'm seeing some console errors (but with all passing tests), looked into it briefly and it seems like it could be a need for a mock config or something? 🤔 

Error:

image


Call site:
image


The remaining console errors look like issues with the tests/code and can be fixed in a follow up, I reckon

**Comment by chihsuan** [comment]
> I'm seeing some console errors (but with all passing tests), looked into it briefly and it seems like it could be a need for a mock config or something? 🤔
> 
> Error:
> 
> image
> 
> Call site: image
> 
> The remaining console errors look like issues with the tests/code and can be fixed in a follow up, I reckon

Yea, good catch. Thanks @rjchow! Mocked `@automattic/calypso-config` in bea155792687826321d80ba0cec243f4e46c24f8.

**Review by rjchow — APPROVED** [review]
Looks great! Thanks!

File: tests/setup/setup-window-globals.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Check if we're in a JSDOM test or not
```
Comment (daledupreez) [inline tests/setup/setup-window-globals.js:1]: Minor note: is this meant to be a TODO?

**Review by daledupreez — APPROVED** [review]
Thanks for tackling this! I had started on this late yesterday, but hadn't made much progress.

Tested well for me, which is _way_ better than before your changes!

File: tests/setup/setup-window-globals.js (hunk 1 above)
Comment (chihsuan) [reply tests/setup/setup-window-globals.js:1]: It's just a doc comment for `global.window` check.

**Pull request merged**